src/pages/storage/StoragePoolSize.tsx

```diff
@@ -1,38 +1,89 @@
 import { FC } from "react";
-import { fetchStoragePoolResources } from "api/storage-pools";
-import { useQuery } from "@tanstack/react-query";
-import { queryKeys } from "util/queryKeys";
-import type { LxdStoragePool } from "types/storage";
+import type {
+  LxdClusteredStoragePoolResources,
+  LxdStoragePool,
+} from "types/storage";
 import { humanFileSize } from "util/helpers";
 import Meter from "components/Meter";
+import { useClusteredStoragePoolResources } from "context/useStoragePools";
+import { isClusteredServer } from "util/settings";
+import { useSettings } from "context/useSettings";
+import { useQuery } from "@tanstack/react-query";
+import { queryKeys } from "util/queryKeys";
+import { fetchStoragePoolResources } from "api/storage-pools";
+import ResourceLink from "components/ResourceLink";
 
 interface Props {
   pool: LxdStoragePool;
   hasMeterBar?: boolean;
 }
 
 const StoragePoolSize: FC<Props> = ({ pool, hasMeterBar }) => {
+  const { data: clusteredPoolResources = [] } =
+    useClusteredStoragePoolResources(pool.name);
+  const { data: settings } = useSettings();
+  const sizeSpecificDrivers = ["zfs", "btrfs", "dir", "lvm"];
+  const hasMemberSpecificSize =
+    sizeSpecificDrivers.includes(pool.driver) && isClusteredServer(settings);
+
   const { data: resources } = useQuery({
     queryKey: [queryKeys.storage, pool.name, queryKeys.resources],
     queryFn: () => fetchStoragePoolResources(pool.name),
+    enabled: !hasMemberSpecificSize,
   });
+  const isClustered = isClusteredServer(settings);
+  const poolsToRender = hasMemberSpecificSize
+    ? clusteredPoolResources
+    : [resources];
```

Comment:
```suggestion
  const resourceList = hasMemberSpecificSize
    ? clusteredPoolResources
    : [poolResources];
```
Maybe a bit better naming?